src/server/client.ts

```diff
@@ -528,7 +528,32 @@ namespace ts.server {
         }
 
         getDocumentHighlights(fileName: string, position: number): DocumentHighlights[] {
-            throw new Error("Not Implemented Yet.");
+            let { line, offset } = this.positionToOneBasedLineOffset(fileName, position);
+            let args: protocol.FileLocationRequestArgs = { file: fileName, line, offset };
+
+            let request = this.processRequest<protocol.DocumentHighlightsRequest>(CommandNames.DocumentHighlights, args);
+            let response = this.processResponse<protocol.DocumentHighlightsResponse>(request);
+
+            let _self = this;

```

Comment:
It's just slightly annoying. I'd still change to `self` though if that doesn't conflict with anything